### PR TITLE
feat: stake period

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: contracts/StakingPool.sol [hunk 1]
```diff
@@ -1,43 +1,99 @@
...
+  uint public stakingBlockLimit;
+  uint public blockToCheckBalance;
 
-  constructor (address tokenAddress) public ERC20Detailed("TellerStatus", "TSNT", 18) {
-    token = IERC20(tokenAddress);
+  /**
+   * @param _tokenAddress SNT token address
+   * @param _stakingPeriodLen Number of block that represents the period when Staking is available
+   */
+  constructor (address _tokenAddress, uint _stakingPeriodLen) public ERC20Detailed("TellerStatus", "TSNT", 18) {
```
Comment (jrainville) [inline contracts/StakingPool.sol]: Do we still want to call the token TellerStatus? I think now it's going to be more generic no?

File: contracts/StakingPool.sol [hunk 2]
```diff
@@ -1,43 +1,99 @@
...
+    return wdiv(_value, wdiv(rate, 1000000000000000000));
   }
 
-  function deposit (uint256 amount) public payable {
-    _deposit(msg.sender, amount);
+  /**
+   * @notice Determine max amount that can be staked
+   * @return Max amount to stake
+   */
+
```
Comment (jrainville) [inline contracts/StakingPool.sol]: There's an extra line
```suggestion
```

File: contracts/StakingPool.sol (hunk 1 above)
Comment (richard-ramos) [reply contracts/StakingPool.sol]: You're right. 
Any proposals?    "SDT" -> "Status DAO Token"? 
cc: @iurimatias

Comment (jrainville) [reply contracts/StakingPool.sol]: That's a good name IMO (though a bit similar to STD lol)

**Pull request merged**